### DISCUSSION TITLE
How to retrieve the emailaddress of a user

### DISCUSSION THREAD

**Discussion by skeutmeier** [codeigniter4/shield#668] [Q&A] [answered]
Hi,
Maybe I overlook a thing, but I cannot figure out how to retrieve the emailaddress of a user by its ID.

I want to send an email from my Backend, so I do know the users ID, and I want to send the user an Email. How can I retrieve the Emailaddress (or any other protected attribute)?

Thank you

**Comment by datamweb** [comment, marked as answer]
Hi, try 

```php
$users = model('UserModel');
$user  = $users->findById(1);
echo ($user->email);
```

**Reply by drumard** [reply]
Did you get this to work? I get a null value for the email address.

When I run this code:
```
        $users = model('UserModel');
        $user  = $users->findById(1);
        echo "username: ".$user->username.", email: |".$user->email."|\n";
```
I get the following output:

```
CodeIgniter v4.3.5 Command Line Tool - Server Time: 2023-06-06 00:38:51 UTC+00:00

username: chair, email: ||
```
Checking the "users" database table there's no email address field nor can I find it in any other table.

Thanks

**Reply by kenjis** [reply]
Email addresses are stored in `auth_identities` like this:
| id | user\_id | type | name | secret | secret2 | expires | extra | force\_reset | last\_used\_at | created\_at | updated\_at |
| :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- |
| 1 | 1 | email\_password | null | admin@example.jp | $2y$10$N... | null | null | 0 | 2023-04-19 06:32:15 | 2023-04-14 00:17:00 | 2023-04-19 06:32:15 |


See https://codeigniter4.github.io/shield/concepts/#user-identities
